packages/block-editor/src/hooks/background.js

```diff
@@ -450,9 +459,21 @@ function BackgroundSizePanelItem( {
 	const updateBackgroundSize = ( next ) => {
 		// When switching to 'contain' toggle the repeat off.
 		let nextRepeat = repeatValue;
+		let nextPosition = positionValue;
 
 		if ( next === 'contain' ) {
 			nextRepeat = 'no-repeat';
+			if ( nextPosition === undefined ) {
+				nextPosition = 'center';
+			}
+		}
+
+		if ( next !== 'contain' && nextPosition === 'center' ) {
+			nextPosition = undefined;
+		}
+
+		if ( next === 'cover' ) {
```

Comment:
This is a new addition, and not _really_ required.

I noticed that background-repeat persisted when switching from contain/auto back to cover.

Is this intentional? If so, I can remove this if statement

Reply:
I don't think it was intentional. Clearing out the repeat value when toggling sounds fine to me.